### PR TITLE
sql/plan: speed up InnerJoin iterator

### PR DESCRIPTION
**Pull request by erizocosmico**
```
pkg: gopkg.in/src-d/go-mysql-server.v0/sql/plan
BenchmarkInnerJoin/inner_join-4         	   50000	     34188 ns/op	   11415 B/op	     169 allocs/op
BenchmarkInnerJoin/cross_join_with_filter-4         	   30000	     41608 ns/op	   12119 B/op	     181 allocs/op
PASS
ok  	gopkg.in/src-d/go-mysql-server.v0/sql/plan	3.795s
```

### CODE REVIEW

File: sql/plan/innerjoin.go [hunk 1]
```diff
@@ -127,3 +125,74 @@ func (j *InnerJoin) TransformExpressions(f sql.TransformExprFunc) (sql.Node, err
...
+			i.r = nil
+			i.leftRow = nil
+			continue
+		}
+
+		if err != nil {
+			return nil, err
+		}
+
+		var row sql.Row
```
Comment (kuba--) [inline sql/plan/innerjoin.go]: actually it looks like we can just write:
```
i.leftRow = append(i.leftRow, rightRow...)
```
am I wrong?

Comment (kuba--) [reply sql/plan/innerjoin.go]: and eval at the end

**Comment by erizocosmico** [comment]
Sure @ajnavarro

File: sql/plan/innerjoin.go (hunk 1 above)
Comment (erizocosmico) [reply sql/plan/innerjoin.go]: We cannot append rightRow to leftRow.

When we iterate we iterate the left side only once and for each left row, we get many right rows.

So the result row is Left row + Right row.

If we do 
```i.leftRow = append(i.leftRow, rightRow...)```

Left row would change each iteration, which is not what we want.

**Comment by erizocosmico** [comment]
Done

**Pull request merged**